src/test/java/stacks/StacksIT.java

```diff
@@ -0,0 +1,143 @@
+package stacks;
+
+import static org.junit.Assert.assertEquals;
+
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+
+import org.junit.AfterClass;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.openqa.selenium.By;
+import org.openqa.selenium.JavascriptExecutor;
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.chrome.ChromeDriver;
+import org.openqa.selenium.chrome.ChromeOptions;
+
+import io.github.bonigarcia.wdm.WebDriverManager;
+import java.util.List;
+
+public class StacksIT {
+    private static WebDriver driver;
+    private static String baseUrl = "https://localhost:9443/instance/stacks/";
+    private static JavascriptExecutor js;
+
+    @BeforeClass
+    public static void setupClass() throws IOException {
+        // Manages the browser driver binary for us
+        WebDriverManager.chromedriver().setup();
+        ChromeOptions options = new ChromeOptions();
+        options.addArguments("--ignore-certificate-errors", "--no-sandbox", "--disable-dev-shm-usage", "--disable-gpu");
+        options.setHeadless(true);
+        driver = new ChromeDriver(options);
+        driver.get(baseUrl);
+
+        js = (JavascriptExecutor) driver;
+
+        String stacksJSON = new String(Files.readAllBytes(Paths.get("src", "test", "resources", "stacks.json")),
+                StandardCharsets.UTF_8);
+
+        // execute javascript functions to set mock data
+        js.executeScript("updateStackView(JSON.parse(arguments[0]));", stacksJSON);
+    }
+
+    @AfterClass
+    public static void teardown() {
+        if (driver != null) {
+            driver.quit();
+        }
+    }
+
+    @Test
+    public void hasCorrectTitleIT() {
+        String expectedTitle = "Stacks";
+        String actualTitle = driver.getTitle();
+        assertEquals("has correct title", expectedTitle, actualTitle);
+    }
+
+    @Test
+    public void stacksTableHasCorrectNumberOfHeadersIT() {
```

Comment:
nice job with the tests, just some renaming comments they're starting to get a little long lol

`hasCorrectNumStackHeadersIT`